common/source/docs/common-brushless-escs.rst

```diff
@@ -36,18 +47,97 @@ The DShot ESC protocol's key advantages are:
 - very high protocol frame rates are supported
 
-While DShot is available on many ESCs, those with BLHeli32 capability, offer the most performance and often, additional features such as ESC telemetry. In addition, ArduPilot provides special programming capability for BLHeli32 ESCs via its passthrough feature (See :ref:`common-dshot` section.)
+While DShot is available on many ESCs, those with BLHeli32 setup capability, offer the most performance and often, additional features such as ESC telemetry. In addition, ArduPilot provides special programming capability for BLHeli32 ESCs via its passthrough feature (See :ref:`common-blheli32-passthru` section.)
 
 .. note::
    Recently there is a growing number of proprietary and non-proprietary 16 / 32 bit ESCs with firmware that support DShot and other digital ESC protocols, but not BLHeli_32-specific features like passthrough and telemetry. See your ESC's manual for further detail on supported features.
 
-Where to buy
-============
+Configuration
+=============
+
+Protocol
+--------
+The ESC control protocol is selected by the :ref:`MOT_PWM_TYPE <MOT_PWM_TYPE>` parameter on Copter and Rover, or :ref:`Q_M_PWM_TYPE <Q_M_PWM_TYPE>` on QuadPlanes for copter function motors. Bi-directional DShot is selected by selecting DShot in :ref:`MOT_PWM_TYPE <MOT_PWM_TYPE>` and then selecting the required channels through :ref:`SERVO_BLH_BDMASK <SERVO_BLH_BDMASK>`.
+
+.. note:: The autopilot should be re-booted after changing the protocol type.
+
+ On Plane, other motors use Normal (PWM) protocol. However, in Plane, other motors, like the traditional fixed wing's main motor or Dual Motor Tailsitters (SERVOn_FUNCTION = 70 throttle, 73 throttle left and / or 74 throttle right), can be changed to another protocol using the :ref:`SERVO_BLH_MASK<SERVO_BLH_MASK>` parameter to specify the output number of the motor together with the :ref:`SERVO_BLH_OTYPE<SERVO_BLH_OTYPE>` parameter to select the protocol of these motors independently of that selected for the copter function motors. 
+
+.. note:: All mask-based configuration can only be changed at a PWM group level, please consult the documentation for your flight controller to ascertain which outputs are on different groups. See :ref:`Autopilot Outputs<output-groups>` section below.
+
+.. warning:: Be sure your ESC can support the configuration you select for it. Damage can occur otherwise. This includes frame rates discussed below. Also be careful when switching between digital and analogue output types without re-calibrating ESCs as this can lead to uncommanded motor output.
+
+For larger aircraft with longer cable runs using DShot ESC protocol, we recommend using the lowest baud rate, DShot150, as it is the most reliable protocol (lower baudrates are less susceptible to noise on cables). Higher rates can be more susceptible to noise but also tie up the allocated DMA channel for a shorter period so can be beneficial on flight controllers with a lot of DMA sharing.
```

Comment:
I think all this DShot advice should go up in the DShot section instead of being down here.

Reply:
done